src/main/scala/com/cognite/sdk/scala/v1/resources/sequenceRows.scala

```diff
@@ -0,0 +1,265 @@
+                case Some(items) => (items.columns.toList, items.rows)
+                case None => (Seq.empty, Seq.empty)
+              }
+          }
+      }
+
+  def queryByExternalId(
+      externalId: String,
+      inclusiveStart: Long,
+      exclusiveEnd: Long
+  ): F[(Seq[SequenceColumnId], Seq[SequenceRow])] =
+    queryByExternalIdWithLimitAndMaybeColumns(externalId, inclusiveStart, exclusiveEnd)
+
+  def queryByExternalIdWithLimit(
+      externalId: String,
+      inclusiveStart: Long,
+      exclusiveEnd: Long,
+      limit: Int
+  ): F[(Seq[SequenceColumnId], Seq[SequenceRow])] =
+    queryByExternalIdWithLimitAndMaybeColumns(
+      externalId,
+      inclusiveStart,
+      exclusiveEnd,
+      limit = limit
+    )
+
+  def queryByExternalIdWithColumns(
+      externalId: String,
+      inclusiveStart: Long,
+      exclusiveEnd: Long,
+      columns: Seq[String]
+  ): F[(Seq[SequenceColumnId], Seq[SequenceRow])] =
+    queryByExternalIdWithLimitAndMaybeColumns(
+      externalId,
+      inclusiveStart,
+      exclusiveEnd,
+      columns = Some(columns)
+    )
+
+  def queryByExternalIdWithColumnsAndLimit(
+      externalId: String,
+      inclusiveStart: Long,
+      exclusiveEnd: Long,
+      columns: Seq[String],
+      limit: Int
+  ): F[(Seq[SequenceColumnId], Seq[SequenceRow])] =
+    queryByExternalIdWithLimitAndMaybeColumns(
+      externalId,
+      inclusiveStart,
+      exclusiveEnd,
+      limit,
+      Some(columns)
+    )
+
+  private def queryByExternalIdWithLimitAndMaybeColumns(
+      externalId: String,
+      inclusiveStart: Long,
+      exclusiveEnd: Long,
+      limit: Int = 100,
+      columns: Option[Seq[String]] = None
+  ): F[(Seq[SequenceColumnId], Seq[SequenceRow])] =
+    requestSession
+      .sendCdf { request =>
+        request
+          .post(uri"$baseUri/list")
+          .body(
+            Items(
+              Seq(
+                SequenceRowsQueryByExternalId(
+                  externalId,
+                  inclusiveStart,
+                  exclusiveEnd,
+                  limit,
+                  columns
+                )
+              )
+            )
+          )
+          .response(asJson[Either[CdpApiError, Items[SequenceRowsResponse]]])
+          .mapResponse {
+            case Left(value) => throw value.error
+            case Right(Left(cdpApiError)) => throw cdpApiError.asException(uri"$baseUri/list")
+            case Right(Right(value)) =>
+              value.items.headOption match {
+                case Some(items) => (items.columns.toList, items.rows)
+                case None => (Seq.empty, Seq.empty)
+              }
+          }
+      }
+}
+
+object SequenceRows {
+  implicit val instantEncoder: Encoder[Instant] = Encoder.encodeLong.contramap(_.toEpochMilli)
+  implicit val instantDecoder: Decoder[Instant] = Decoder.decodeLong.map(Instant.ofEpochMilli)
+
+  implicit val cogniteIdEncoder: Encoder[CogniteId] = deriveEncoder
```

Comment:
Could these have been created with `io.circe.generic.auto._` instead?

Reply:
Yes, but we intentionally avoid auto derivation because the compile time increases too much (it more than doubled when I ran my experiments) compared to using the much faster https://github.com/circe/circe-derivation with `deriveEncoder/Decoder`, and I don't think it's worth it just to remove this boilerplate.

With this we also don't need the users to import `io.circe.generic.auto._` before using the SDK, not sure if that would have been possible to get around in some other way.